ssz/src/main/java/tech/pegasys/artemis/util/backing/cache/Cache.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.util.backing.cache;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Cache
+ *
+ * @param <K> type of keys
+ * @param <V> type of values
+ */
+public interface Cache<K, V> {
```

Comment:
Yep ,that's currently `TODO`. 
When working on SSZ refactor I'm planning to eliminate `util` -> `ssz` module dependency and return back `cache` package to `util` module (since it looks pretty  generic)